chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -214,10 +242,58 @@
         Chef::Log.info("#{name} usurped #{res[1].join(", ")} routes from #{bound_if}") unless res[1].empty?
       end
     end
+
+    execute "Neutron: add #{bound_if} to #{name}" do
+      command "ovs-vsctl del-port #{name} #{bound_if} ; ovs-vsctl add-port #{name} #{bound_if}"
+      not_if "ovs-dpctl show system@#{name} | grep -q #{bound_if}"
+    end
     source = ::Nic.new(bound_if)
     # filter out cluster VIPs that a currently assigned to the interface
     addresses = source.addresses.select { |address| my_addresses.include? address }
     routes = source.routes
+
+    # Try to bring the cluster back up.
+    if (node[:crowbar_wall][:network][:nets]["admin"].last == bound_if &&
+        neutron_l3_ha && node[:corosync][:transport] == "udp" )
+
+      service "start openais after network reconfiguration" do
+        service_name node[:corosync][:platform][:service_name]
+        action [:start]
+        not_if { addresses.empty? and routes.empty? }
+      end
+
+      crowbar_pacemaker_sync_mark "sync-neutron-l3_after_network_reconfig"
+
+      # FIXME: copied from pacemaker::default, should probably move this into a
+      # Helper
+      ruby_block "wait for cluster to be online" do
+        block do
+          require 'timeout'
+          begin
+            Timeout.timeout(60) do
+              cmd = "crm_mon -1 | grep -qi online"
```

Comment:
Actually I think it would be better to wait for the whole cluster. Should be possible by adding and addtionally sync-mark here. Provided that it's safe to assume the cluster is back up when every node is able to talk to it's local pacemaker + we have a sync mark after that.